types/react/index.d.ts

```diff
@@ -2935,6 +3117,14 @@ type ReactManagedAttributes<C, P> = C extends { propTypes: infer T; defaultProps
             ? Defaultize<P, D>
             : P;
 
+// We can't recurse forever because `type` can't be self-referential;
+// let's assume it's reasonable to do a single React.lazy() around a single React.memo() / vice-versa
+export type JSXLibraryManagedAttributes<C, P> = C extends React.MemoExoticComponent<infer T> | React.LazyExoticComponent<infer T>
```

Comment:
Why not name this ReactLibraryManagedAttributes / put this in the React namespace, like you did with IntrinsicElements?

Reply:
Good question for which I have no good answer 😅 I believe I have applied your suggestion.